### DISCUSSION TITLE
Problems access instancve ports from browser on local machine

### DISCUSSION THREAD

**Discussion by ilya1st** [lima-vm/lima#1018] [Q&A]
is there way make all port of VM accessible on local machine?
E.g. I expose ports  of my application but they are accessible only inside vm

It's not good

**Comment by ilya1st** [comment]
and yes there are problems with kubernetes "opening" nodeports  sockets via iptables  layer^ that's why sockerts are not visible via sockstat that's why lima-guestagent daemon cannot see them.
<img width="1372" alt="image" src="https://user-images.githubusercontent.com/9489084/184597753-dd7c24ef-db43-431a-b9d6-b3dd8f7a46ad.png">

**Reply by AkihiroSuda** [reply]
I guess we need to adjust the regexp here to support kube NodePorts ?


https://github.com/lima-vm/lima/blob/4edd35381c48e33721f54868d75408a4e6b41c8c/pkg/guestagent/iptables/iptables.go#L20-L34

**Reply by AkihiroSuda** [reply]
Opened an issue
- https://github.com/lima-vm/lima/issues/1021

**Reply by ilya1st** [reply]
@AkihiroSuda may be. I use kubernetes 1.24 and in uses nftables  for forwarding not iptables
In nft ruleset it looks like:
<img width="956" alt="image" src="https://user-images.githubusercontent.com/9489084/184868773-f60730e6-1808-4019-9d5b-76d153ff930b.png">

iptables cannot see nft rules
